crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -284,13 +284,16 @@ async fn app_validation_workflow_inner(
 
                     warrant_op_hashes.push((warrant_op.to_hash(), warrant_op.dht_basis().clone()));
 
-                    workspace
+                    if let Err(err) = workspace
                         .authored_db
                         .write_async(move |txn| {
                             warn!("Inserting warrant op");
                             insert_op_authored(txn, &warrant_op)
                         })
-                        .await?;
+                        .await
```

Comment:
This was failing silently and not bubbled up.

Reply:
Should where this bubbles up to not also be logging then?

Reply:
It does not bubble up, that's why I didn't find the problem until I didn't try the result any more and logged it. Logging it outputs
```
Err(Sql(SqliteFailure(Error { code: ConstraintViolation, extended_code: 787 }, Some("FOREIGN KEY constraint failed"))))
```
I have no idea why it doesn't bubble up, it's not inside of a closure.